jsonapi/fixtures_test.go

```diff
@@ -361,86 +449,110 @@ func (q Question) GetReferencedStructs() []MarshalIdentifier {
 
 type Identity struct {
 	ID     int64    `json:"-"`
+	LID    int64    `json:"-"`
 	Scopes []string `json:"scopes"`
 }
 
-func (i Identity) GetID() string {
-	return fmt.Sprintf("%d", i.ID)
+func (i Identity) GetID() Identifier {
+	id := Identifier{ID: fmt.Sprintf("%d", i.ID), Name: "identities"}
+	if i.LID != 0 {
+		id.LID = fmt.Sprintf("%d", i.LID)
+	}
+	return id
 }
 
-func (i *Identity) SetID(ID string) error {
-	var err error
-	i.ID, err = strconv.ParseInt(ID, 10, 64)
-	return err
+func (i *Identity) SetID(ID Identifier) error {
+	if ID.ID != "" {
+		id, err := strconv.Atoi(ID.ID)
+		if err != nil {
+			return err
+		}
+		i.ID = int64(id)
+	}
+	if ID.LID != "" {
+		lid, err := strconv.Atoi(ID.LID)
+		if err != nil {
+			return err
+		}
+		i.LID = int64(lid)
+	}
+
+	return nil
 }
 
 type Unicorn struct {
 	UnicornID int64    `json:"unicorn_id"` // annotations are ignored
 	Scopes    []string `json:"scopes"`
 }
 
-func (u Unicorn) GetID() string {
-	return "magicalUnicorn"
+func (u Unicorn) GetID() Identifier {
+	return Identifier{ID: "magicalUnicorn", LID: "", Name: "unicorns"}
 }
 
 type NumberPost struct {
 	ID             string `json:"-"`
+	LID            string `json:"-"`
 	Title          string
 	Number         int64
 	UnsignedNumber uint64
 }
 
-func (n *NumberPost) SetID(ID string) error {
-	n.ID = ID
+func (n NumberPost) GetID() Identifier{
+	return Identifier{ID: n.ID, LID: n.LID, Name: "number-posts"}
+}
+
+func (n *NumberPost) SetID(ID Identifier) error {
+	n.ID = ID.ID
+	n.LID = ID.LID
 	return nil
 }
 
 type SQLNullPost struct {
 	ID     string      `json:"-"`
+	LID    string      `json:"-"`
 	Title  zero.String `json:"title"`
 	Likes  zero.Int    `json:"likes"`
 	Rating zero.Float  `json:"rating"`
 	IsCool zero.Bool   `json:"isCool"`
 	Today  zero.Time   `json:"today"`
 }
 
-func (s SQLNullPost) GetID() string {
-	return s.ID
+func (s SQLNullPost) GetID() Identifier {
+	return Identifier{ID: s.ID, LID: s.LID, Name: "sqlNullPosts"}
```

Comment:
No need to define Name, since GetName was not implemented before

Reply:
removed in other cases when GetName was not implemented initially, here (SQLNulPost) it actually was here before my changes